drivers/adc/ad7746/iio_ad7746.c

```diff
@@ -0,0 +1,714 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <math.h>
+#include "error.h"
+#include "delay.h"
+#include "iio.h"
+#include "iio_ad7746.h"
+#include "util.h"
+#include "ad7746.h"
+
+static int32_t _ad7746_read_register2(struct ad7746_dev *dev, uint32_t reg,
+				      uint32_t *readval)
+{
+	return ad7746_reg_read(dev, reg, (uint8_t *)readval, 1);
+}
+
+static int32_t _ad7746_write_register2(struct ad7746_dev *dev, uint32_t reg,
+				       uint32_t writeval)
+{
+	return ad7746_reg_write(dev, reg, (uint8_t *)writeval, 1);
+}
+
+static inline bool _capdiff(struct ad7746_cap *cap1, struct ad7746_cap *cap2)
+{
+	return cap1->capchop != cap2->capchop ||
+	       cap1->capdiff != cap2->capdiff ||
+	       cap1->capen != cap2->capen ||
+	       cap1->cin2 != cap2->cin2;
+}
+
+static inline bool _vtdiff(struct ad7746_vt *vt1, struct ad7746_vt *vt2)
+{
+	return vt1->extref != vt2->extref ||
+	       vt1->vtchop != vt2->vtchop ||
+	       vt1->vten != vt2->vten ||
+	       vt1->vtmd != vt2->vtmd ||
+	       vt1->vtshort != vt2->vtshort;
+}
+
+static inline bool _configdiff(struct ad7746_config *c1,
+			       struct ad7746_config *c2)
+{
+	return c1->md != c2->md ||
+	       c1->capf != c2->capf ||
+	       c1->vtf != c2->vtf;
+}
+
+/* Values are Update Rate (Hz), Conversion Time (ms) + 1*/
+static const uint8_t ad7746_vt_filter_rate_table[][2] = {
+	{50, 20 + 1}, {31, 32 + 1}, {16, 62 + 1}, {8, 122 + 1},
+};
+
+static const uint8_t ad7746_cap_filter_rate_table[][2] = {
+	{91, 11 + 1}, {84, 12 + 1}, {50, 20 + 1}, {26, 38 + 1},
+	{16, 62 + 1}, {13, 77 + 1}, {11, 92 + 1}, {9, 110 + 1},
+};
+
+static int ad7746_store_cap_filter_rate_setup(struct ad7746_dev *chip,
+		int val)
+{
+	unsigned int i;
```

Comment:
maybe use `stdint` typedefs here.
Check entire code.

Reply:
good point, completely forgot about this

Reply:
so i've done some modifications in this regard but couldn't do all of them because in someplaces i used int because existing iio_types.h uses types like that. I think in those cases we should chean iio up. Until then I fixed only the independent int / unsigned int occurences.